### PR TITLE
Feature/api

### LINKED ISSUE #10: API Calls - all movies
Refactor the application to use actual data for all movies from the database instead of mock data

### LINKED ISSUE #11: API Calls - single movie
Refactor the application to use actual data for retrieving a single movie's details from the database instead of mock data

### LINKED ISSUE #12: API Calls - error handling
As a user, when the server returns a 500 error, the user will see proper error handling.

### PR DESCRIPTION
**Pull request by nbalmoori**
## Summary:
Refactored the application to use actual data for all movies and single movie details from the database instead of using mock data.

Additionally, we added new error handling features so that if the server returns a 500 level error when retrieving all movie data or single movie details, the application will display an error message for the user.

## Type of Changes:
- [X] New feature
- [ ] Bug fix
- [X] Refactor

## Where are the Changes:
Fetch calls are located in `App.js`.  Error handling updates to JSX are located in `MovieContainer.js` and `MovieDetails.js`.

## How was this tested:
Tested using npm start in the browser.

## What are the relevant tickets:
Close #10, Close #11, Close #12

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,19 +4,40 @@ import Nav from './Nav'
...
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+      .then(response => {
+        if (!response.ok) {
+          throw Error()
+        } else {
+          return response.json()
+        }
+      })
+      .then(data => this.setState({currentMovie: data.movie}))
+      .catch(err => this.setState({currentMovie: "error"}))
```
Comment (JoeRupp) [inline src/App.js:40]: We should ask if we should move our fetch calls to their own file and then call them in App? Similar to what we've done in the past. Not sure what the "proper" way is in React.

File: src/MovieDetails.js [hunk 2]
```diff
@@ -2,21 +2,31 @@ import React from 'react';
...
+      <div className='movieDetails'>
+        <img className='movieBackdrop' src= {currentMovie.backdrop_path}/>
+        <div className='movieInfo'>
+          <button className='homeButton' onClick={() => displayHomeScreen()}>Back</button>
+          <h1 className='movieTitle'>{currentMovie.title}</h1>
+          <div className='movieDescription'>
+            <p>{currentMovie.tagline.toUpperCase()}</p>
+            <p>{currentMovie.overview}</p>
+          </div>
+          <p className='movieFacts'>{`Rating: ${currentMovie.average_rating}/10 | Genres: ${currentMovie.genres.join(",")} | Time: ${currentMovie.runtime} mins | Release Date: ${currentMovie.release_date} | Budget: $${currentMovie.budget.toLocaleString()} | Revenue: $${currentMovie.revenue.toLocaleString()}`} </p>
```
Comment (JoeRupp) [inline src/MovieDetails.js:24]: Nice work with this conditional render here!

**Comment by JoeRupp** [comment]
Looks great! Biggest question I have is around where to API calls (see comment in code). We can ask and move later if needed.

File: src/App.js [hunk 3]
```diff
@@ -4,19 +4,40 @@ import Nav from './Nav'
...
 import './App.css';
-import movieData from './movieData';
-import movieDetails from './movieDetailData';
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movieList: movieData.movies,
+      movieList: [],
       currentMovie: ""
```
Comment (kaylagordon) [inline src/App.js:12]: You should be able to remove this piece of state after implementing Router

File: src/App.js [hunk 4]
```diff
@@ -4,19 +4,40 @@ import Nav from './Nav'
...
     }
   }
 
+  componentDidMount = () => {
+    fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
```
Comment (kaylagordon) [inline src/App.js:17]: Consider moving your API calls to a separate file like we did in M2

File: src/App.js [hunk 5]
```diff
@@ -4,19 +4,40 @@ import Nav from './Nav'
...
+      .then(response => {
+        if (!response.ok) {
+          throw Error()
+        } else {
+          return response.json()
+        }
+      })
+      .then(data => this.setState({movieList: data.movies}))
+      .catch(err => console.log(err))
```
Comment (kaylagordon) [inline src/App.js:26]: Consider doing some error handling that the user can see, rather than just console logging

File: src/App.js (hunk 1 above)
Comment (kaylagordon) [reply src/App.js:40]: Yes!

File: src/App.js [hunk 6]
```diff
@@ -4,19 +4,40 @@ import Nav from './Nav'
...
+  }
+
+
   chooseMovie = (id) => {
-    this.setState({ currentMovie: movieDetails.movie })
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
```
Comment (kaylagordon) [inline src/App.js:31]: Does your App component need all of this data? Or could you make this fetch in a different component? (You might find that yes, in fact you DO need this in App, I just want you to think about it.)

**Pull request merged**